### DISCUSSION TITLE
Offline support of Mobile client not working

### DISCUSSION THREAD

**Discussion by StfnGe** [dani-garcia/vaultwarden#4383] [Q&A] [answered]
### Offline support of Mobile client not working
I want to use Vaultwarden as my password manager. When i am in my own network i works flawlessly, but as soon as i am disconnected from my network or in offline mode on my smartphone the database gets disconnected when i open the Bitwarden client on my S24 Ultra (One UI Version 6.1). 

I used the online hosted Bitwarden before in the same way without any issue. In both cases (self hosted/online official Bitwarden) MFA is active.

### Deployment environment

* vaultwarden version: 1.30.3

* Install method: Docker

* Clients used: Android

* Reverse proxy and version: Nginx reverse proxy 

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details: only internaly available

### Steps to reproduce
Login into Android client, works. Go offline, session is ended and need to relogin. Otherwise no entrys will be shown.

### Expected behaviour
Offline access works

### Actual behaviour
Offline access doesn't work

### Troubleshooting data
-

**Comment by BlackDex** [comment]
If your session is ended then you need to login, which is kinda normal i think.
I can use my mobile client just fine when i'm in airplane mode.

**Reply by StfnGe** [reply]
Yes but this shouldn't happen when i just logged in and try to open the client 2 minutes later in offline mode. The session doesnt end this fast .. or does it?

**Comment by jjlin** [comment, marked as answer]
See https://vaultwarden.discourse.group/t/offline-online-access/2298

**Reply by StfnGe** [reply]
Thanks for the reply. I think this could be the issue, as the status is 403 when i am outside of my network. Gonna try to change the response from the reverse proxy.

**Reply by StfnGe** [reply]
Yes this was the issue. I changed the forbidden to an redirect to my homepage when visiting the url from the outside. Instantly worked :-)

Realised with the geo module of Nginx:
```shell
geo $internal {
    default        0;
    MYINTERNALNETWORK/24    1;
}

server {

    if ($internal != 1) {
        rewrite ^ https://www.somedomain.com;
    }

[...]restofconfig
```
